### PR TITLE
Update windows tests

### PR DESCRIPTION
**Pull request by nskaggs**
## Description of change

This change simplifies our windows testing, removing our need for workspace runner and replacing with a simple bash script instead. This also allows us to utilize win32-powershell-openssh, rather than a cygwin ssh. Given the nature of the openssh change, we no longer have a unix-like environment to run in. Instead, the approach has been to dynamically create powershell and batch scripts as needed, copy them to the machine, and then run them via ssh.

## QA steps

The unit test jobs running as part of CI and during pre-check will be vetted before landing.

### CODE REVIEW

**Comment by nskaggs** [comment]
Note, that jenkins didn't fail the run, despite the error in the windows test: http://juju-ci.vapour.ws/job/github-check-merge-juju/1445/artifact/artifacts/windows.log/*view*/

File: acceptancetests/run-windows-job.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+host=$1
+ts=$(date +%s)
+tarfile=''
+echo "Running on $host as payload $ts"
+runcmd "mkdir C:\\Users\\Administrator\\payloads\\$ts"
+
+if [[ $2 == *"tar"* ]]; then
+    tarfile=$2
+    echo "Using $tarfile as tarball"
+elif [ "$2" -eq "$2" ] 2>/dev/null; then
```
Comment (Veebers) [inline acceptancetests/run-windows-job.sh]: I'm not sure what this is checking, this will always be true and the fall-through else will never be used.

File: acceptancetests/run-windows-job.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    scp -i $JUJU_HOME/staging-juju-rsa -oStrictHostKeyChecking=no $payload $host:C:\\Users\\Administrator\\payloads\\
+    #Run test command via ssh
+    ssh $host C:\\Users\\Administrator\\payloads\\$payload
+}
+
+host=$1
+ts=$(date +%s)
+tarfile=''
+echo "Running on $host as payload $ts"
+runcmd "mkdir C:\\Users\\Administrator\\payloads\\$ts"
```
Comment (Veebers) [inline acceptancetests/run-windows-job.sh]: I would prefer these variables to be setup at the to before the function, purely to save having to bounce up and down to see what $ts is within the function.

Also can we have host=$1 for completeness :-)

File: acceptancetests/update-jenkins-slave.bash [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 update_branch() {
     # Branch or pull a branch.
     local_branch=$1
     local_dir="$(basename $local_branch | cut -d ':' -f2)"
     local_path="$HOME/$local_dir"
     if [[ -d $local_path ]]; then
-        # Clear any changes that haven't been commited to the branch.
+        # Clear any changes that have not been commited to the branch.
         (cd $local_path; bzr revert; bzr pull)
+        cd ..
```
Comment (Veebers) [inline acceptancetests/update-jenkins-slave.bash]: Why is this needed, the use of (cd ...) should scope the chwd change so that there is no "cleanup needed."

File: acceptancetests/run-windows-job.sh (hunk 1 above)
Comment (nskaggs) [inline acceptancetests/run-windows-job.sh]: This is an interesting hack to check for digits. It's a bashism. You can see my previous comparison akin to looking for digits in af5fea5, but that falls down since the payload scripts are timestamp files. This ensure the arg is a number, and only a number.

File: acceptancetests/update-jenkins-slave.bash (hunk 3 above)
Comment (nskaggs) [inline acceptancetests/update-jenkins-slave.bash]: Yea, a bit funny. I think this was needed because git operation needed to go into the local directory to perform the operation. It would be nicer to have it cd back to a specific directory, but it is required I believe.

File: acceptancetests/run-windows-job.sh (hunk 2 above)
Comment (nskaggs) [inline acceptancetests/run-windows-job.sh]: Ohh.. I understand your meaning. I think I prefer putting all functions right at the top. But it does make it harder to read. Bash won't let you move things nicely because of how it's interpreted. That said, I think you could put the declarations above the function. I just don't want to split the code. I find that harder / more hidden to read.

**Comment by Veebers** [comment]
$merge$

**Comment by Veebers** [comment]
$$merge$$

**Comment by Veebers** [comment]
$$merge$$
